go-controller/pkg/ovn/base_network_controller_namespace.go

```diff
@@ -174,6 +178,32 @@ func (bnc *BaseNetworkController) syncNamespaces(namespaces []interface{}) error
 	if err != nil {
 		return fmt.Errorf("error in syncing namespaces: %v", err)
 	}
+
+	// remove stale port groups
+	p := func(pg *nbdb.PortGroup) bool {
+		// there is no way to distinguish namespace-owned port group, since its name is just namespace name
```

Comment:
So I would think first if this would be a problem for 4.14. I would say no at first glance, since the chances of us introducing a new PG in 4.14 should be low; but then I think it is going to be long term support, the major architectural change that has had and what we have been capable of back-porting in the past, and I am not sure anymore.

But I guess the correct way of thinking is that this should not be a problem for 4.14.

So I would take this as is with an added big FIXME warning comment here. That way we keep track of it for 4.15 and the warning stays forever in 4.14 when you backport.

Reply:
makes sense, I  added an extra filter condition to exclude port groups with new dbIDs from the predicate.
So I hope we merge dbIDs soon enough so that all new port groups will be created with dbIDs, then if we ever decide to backport them, we will do that with their new dbIDs, and they will be ignored by the predicate anyway.